genesyscloud/integration_custom_auth_action/genesyscloud_integration_custom_auth_action_proxy.go

```diff
@@ -0,0 +1,232 @@
+	action, resp, err := p.integrationsApi.GetIntegrationsAction(actionId, "", true)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return action, resp, nil
+}
+
+// updateCustomAuthActionFn is the implementation for updating an integration custom auth action in Genesys Cloud
+func updateCustomAuthActionFn(ctx context.Context, p *customAuthActionsProxy, actionId string, updateAction *platformclientv2.Updateactioninput) (*platformclientv2.Action, *platformclientv2.APIResponse, error) {
+	action, resp, err := p.integrationsApi.PatchIntegrationsAction(actionId, *updateAction)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return action, resp, nil
+}
+
+// getIntegrationActionTemplateFn is the implementation for getting the integration action template in Genesys Cloud
+func getIntegrationActionTemplateFn(ctx context.Context, p *customAuthActionsProxy, actionId string, fileName string) (*string, *platformclientv2.APIResponse, error) {
+	template, resp, err := p.integrationsApi.GetIntegrationsActionTemplate(actionId, fileName)
+	if err != nil {
+		return nil, resp, err
+	}
+	return template, resp, nil
+}
+
+// getIntegrationTypeFn is the implementation for getting the type of an integration in Genesys Cloud
+func getIntegrationTypeFn(ctx context.Context, p *customAuthActionsProxy, integrationId string) (string, error) {
+	integration, _, err := p.integrationsApi.GetIntegration(integrationId, 1, 1, "", nil, "", "")
+	if err != nil {
+		return "", err
+	}
+
+	return *integration.IntegrationType.Id, nil
+}
+
+// getIntegrationCredentialsTypeFn is the implementation for getting the type of an integration credential in Genesys Cloud
+func getIntegrationCredentialsTypeFn(ctx context.Context, p *customAuthActionsProxy, integrationId string) (string, error) {
+	integrationConfig, _, err := p.integrationsApi.GetIntegrationConfigCurrent(integrationId)
+	if err != nil {
+		return "", err
+	}
+	if integrationConfig.Credentials == nil || len(*integrationConfig.Credentials) == 0 {
+		return "", fmt.Errorf("no credentials set for integration %s", integrationId)
+	}
+
+	basicAuth, found := (*integrationConfig.Credentials)["basicAuth"]
+	if !found {
+		return "", fmt.Errorf("no 'basicAuth' credentials set for integration %s", integrationId)
+	}
+
+	credential, _, err := p.integrationsApi.GetIntegrationsCredential(*basicAuth.Id)
+	if err != nil {
+		return "", err
+	}
+
+	return *credential.VarType.Name, nil
+}
+
+// getIntegrationByNameFn is the implementation for getting a Genesys Cloud Integration by name
+func getIntegrationByNameFn(ctx context.Context, p *customAuthActionsProxy, integrationName string) (*platformclientv2.Integration, bool, error) {
+	var foundIntegration *platformclientv2.Integration
+
+	const pageSize = 100
+	for pageNum := 1; ; pageNum++ {
+		integrations, _, err := p.integrationsApi.GetIntegrations(pageSize, pageNum, "", nil, "", "")
```

Comment:
We're moving towards a different pattern where we make an initial GET call, save the pageCount, then enter the for loop and iterate until `pageNum <= pageCount`
This is because, although most will return a 200, some endpoints will return an error if you set pageNum to a number greater than pageCount.